python/pyarrow/_flight.pyx

```diff
@@ -73,6 +100,45 @@ class CertKeyPair(_CertKeyPair):
     """A TLS certificate and key for use in Flight."""
 
 
+cdef class FlightError(Exception):
```

Comment:
Should we inherit from ArrowException here rather than below?

Reply:
I had to do this dance to make Cython happy: ArrowException isn't cdef, so inheriting from it from a cdef class doesn't work.

Reply:
Ok, that's fine with me.